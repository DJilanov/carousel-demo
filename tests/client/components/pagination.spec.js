const Vue = require('vue');
const utils = require('../utils');

const Carousel = require('../../../src/components/carousel/Carousel.vue');
const Slide = require('../../../src/components/carousel/Slide.vue');

describe('Pagination', () => {
  let vm;
  let carouselInstance;
  let $pagination;

  beforeEach(() => {
    vm = new Vue({
      el: document.createElement('div'),
      render: (h) => h(Carousel, { props: { paginationEnabled: true, perPage: 1 } }, [h(Slide), h(Slide), h(Slide), h(Slide), h(Slide)]),
    });
    carouselInstance = vm.$children[0];
    $pagination = vm.$el.querySelector('.carousel-pagination');
  });

  it('should mount successfully', () => {
    expect($pagination).toBeDefined();

    return utils.expectToMatchSnapshot(vm);
  });

  it('should render a next button', () => {
    expect(vm.$el.querySelector('svg.carousel-next-button')).toBeDefined();

    return utils.expectToMatchSnapshot(vm);
  });

  it('should render a prev button', () => {
    expect(vm.$el.querySelector('svg.carousel-back-button')).toBeDefined();

    return utils.expectToMatchSnapshot(vm);
  });
});
