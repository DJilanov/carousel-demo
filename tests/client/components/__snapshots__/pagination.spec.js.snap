exports[`Navigation should mount successfully 1`] = `
".carousel-wrapper
  .carousel-inner(role=\'listbox\', style=\'visibility: hidden;\')
    .carousel-slide
    .carousel-slide
|  
.carousel-pagination
  ul.carousel-dot-container(role=\'tablist\')
    li.carousel-dot(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'false\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.false(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 0\', tabindex=\'0\', class=\'active:\')
    li.carousel-dot.carousel-dot--active(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'true\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.true(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 1\', tabindex=\'0\', class=\'active:\')
  |  
  svg.flickity-button-icon(viewbox=\'0 0 100 100\', style=\'display: inherit;\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
  |  
  svg.flickity-button-icon.next(viewbox=\'0 0 100 100\', style=\'\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
"
`;

exports[`Navigation should render a next button 1`] = `
".carousel-wrapper
  .carousel-inner(role=\'listbox\', style=\'visibility: hidden;\')
    .carousel-slide
    .carousel-slide
|  
.carousel-pagination
  ul.carousel-dot-container(role=\'tablist\')
    li.carousel-dot(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'false\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.false(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 0\', tabindex=\'0\', class=\'active:\')
    li.carousel-dot.carousel-dot--active(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'true\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.true(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 1\', tabindex=\'0\', class=\'active:\')
  |  
  svg.flickity-button-icon(viewbox=\'0 0 100 100\', style=\'display: inherit;\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
  |  
  svg.flickity-button-icon.next(viewbox=\'0 0 100 100\', style=\'\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
"
`;

exports[`Navigation should render a prev button 1`] = `
".carousel-wrapper
  .carousel-inner(role=\'listbox\', style=\'visibility: hidden;\')
    .carousel-slide
    .carousel-slide
|  
.carousel-pagination
  ul.carousel-dot-container(role=\'tablist\')
    li.carousel-dot(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'false\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.false(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 0\', tabindex=\'0\', class=\'active:\')
    li.carousel-dot.carousel-dot--active(aria-hidden=\'false\', role=\'presentation\', aria-selected=\'true\', style=\'margin-top: 20px;\')
      button.carousel-dot-button.true(type=\'button\', role=\'button\', aria-label=\'\`Item \${index}\`\', title=\'Item 1\', tabindex=\'0\', class=\'active:\')
  |  
  svg.flickity-button-icon(viewbox=\'0 0 100 100\', style=\'display: inherit;\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
  |  
  svg.flickity-button-icon.next(viewbox=\'0 0 100 100\', style=\'\')
    path.arrow(d=\'M 10,50 L 60,100 L 60,95 L 15,50  L 60,5 L 60,0 Z\')
"
`;
